planner/optimize.go

```diff
@@ -32,6 +39,26 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 		return fp, nil
 	}
 
+	var oriHint *bindinfo.HintsSet
+	if stmtNode, ok := node.(ast.StmtNode); ok {
+		oriHint = addHint(sctx, stmtNode)
+	}
+	plan, err := optimize(ctx, sctx, node, is)
+	// Restore the original hint in case of prepare stmt.
+	if oriHint != nil {
+		node = bindinfo.BindHint(node.(ast.StmtNode), oriHint)
+		if err != nil {
+			handleInvalidBindRecord(ctx, sctx, node.(ast.StmtNode))
+		}
+	}
+	if err == nil || oriHint == nil {
```

Comment:
should we log the error when `err != nil && oriHint != nil`?

Reply:
But we already handled the invalid record in this case, is a log necessary here?